test/puppetlabs/trapperkeeper/services/authorization/authorization_service_test.clj

```diff
@@ -0,0 +1,58 @@
+(ns puppetlabs.trapperkeeper.services.authorization.authorization-service-test
+  (:require
+    [clojure.test :refer :all]
+    [clojure.string :as str]
+    [puppetlabs.trapperkeeper.app :refer [get-service]]
+    [puppetlabs.trapperkeeper.services :refer [defservice]]
+    [puppetlabs.trapperkeeper.services.authorization.authorization-service :refer [authorization-service]]
+    [puppetlabs.trapperkeeper.testutils.bootstrap :refer [with-app-with-config]]
+    [puppetlabs.trapperkeeper.testutils.logging :refer [with-test-logging]]
+    [ring.util.response :refer [response]]
+    [schema.test :as schema-test]))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Utilities
+
+(defprotocol EchoReverseService
+  (echo-reverse [this msg]))
+
+; A simple service using authorization via ring middleware
+(defservice
+  echo-reverse-service EchoReverseService [[:AuthorizationService wrap-handler]]
+  (echo-reverse
+    [this msg]
+    (let [handler (fn [req] (response (str msg (str/reverse (:body req)))))]
+      (wrap-handler handler))))
+
+(def minimal-config
+  "Minimal config required to bootstrap AuthorizationService with
+  EchoReverseService as a dependant consumer.
+
+  An empty list representing no rules with an expected behavior of
+  default-deny originating in the authorization library itself."
+  {:authorization {:rules []}})
+
+(def base-request
+  "A basic request to feed into the tests"
+  {:uri "/foo/bar"
+   :request-method :get
+   :remote-addr "127.0.0.1"})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Tests
+
+(deftest ^:integration wrap-handler-test
```

Comment:
I think this is a good test and understand that we're not looking to retest the functionality of the rules translation or `wrap-authorization-check` middleware layer at this level.  It does, however, feel like we could use a couple of more tests here to at least convince ourselves that when a request is handled that the middleware is authorizing against some non-empty rule content from the configuration.  Maybe could just add a couple of extra tests with a single rule where the request being made is denied or allowed as expected?  If there's a good way to mock around the middleware function to avoid having to generate an SSL cert to make the test more system-level like, I'd be fine with that.